packages/flutter_driver/lib/src/driver/driver.dart

```diff
@@ -417,6 +418,33 @@ class FlutterDriver {
     return GetTextResult.fromJson(await _sendCommand(new GetText(finder, timeout: timeout))).text;
   }
 
+  /// Enters `text` into the currently focused text input, such as the
+  /// [EditableText] widget.
+  ///
+  /// This method does not use the operating system keyboard to enter text.
+  /// Instead it emulates text entry by sending events identical to those sent
+  /// by the operating system keyboard.
+  ///
+  /// It is assumed that the widget receiving text input is focused prior to
+  /// calling this method. Typically, a test would activate a widget, e.g. using
+  /// [tap], then call this method.
+  ///
+  /// Example:
+  ///
+  /// ```dart
+  /// test('enters text in a text field', () async {
+  ///  var textField = find.byValueKey('enter-text-field');
+  ///  await driver.tap(textField);  // acquire focus
+  ///  await driver.enterText('Hello!');  // enter text
+  ///  await driver.waitFor(find.text('Hello!'));  // verify text appears on UI
+  ///  await driver.enterText('World!');  // enter another piece of text
+  ///  await driver.waitFor(find.text('World!'));  // verify new text appears
```

Comment:
So, does `enterText` replace the pervious content of the text field? If so, we should maybe document that (and maybe even rename it to `replaceText`)? If not, it would be nice if the example showed that the previous text is still there?

Reply:
Technically the behavior depends on the widget that receives the input, but I think the expectation for text-field-like widgets is that they replace the text. For example, in the case of `EditableText` we replace the entire string. The name "enterText" and the behavior of this method is identical to that of `TestTextInput.enterText`, so I'd like to keep it as is. However, I expanded the dartdoc explaining this behavior in detail.

Bonus: @cbracken just told me that the composing `TextRange` was not necessary to replace the text, so I removed it.

Reply:
@goderbauer Correct - this would be more accurately named `setText` or `replaceText`. The current protocol is such that text update messages are expected to include the full current editing state, not a delta.